hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/StatisticalMetricsTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core.metrics;
+
+import com.hazelcast.jet.Jet;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.core.Vertex;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.test.TestSources;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectInstance;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.Util.entry;
+import static com.hazelcast.jet.core.Edge.between;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
+import static com.hazelcast.jet.core.TestProcessors.ListSource;
+import static com.hazelcast.jet.core.TestProcessors.MockP;
+import static com.hazelcast.jet.core.TestProcessors.MockPMS;
+import static com.hazelcast.jet.core.TestProcessors.MockPS;
+import static com.hazelcast.jet.core.TestProcessors.reset;
+import static com.hazelcast.jet.core.test.JetAssert.assertTrue;
+import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class StatisticalMetricsTest {
+
+    private static final int MEMBER_COUNT = 1;
```

Comment:
We should test on 2 members and check the non-master for execution-related metrics, esp. after fixing `JOB_EXECUTIONS_STARTED`

Reply:
Yes, indeed, I'll add a generic TODO for it and try to do it next week, this shouldn't hinder us in merging in.